src/model/progression.rs

```diff
@@ -20,6 +20,12 @@ impl Progression {
     {
         sep_by(Chord::parser(), spaces()).map(|chords: Vec<_>| Self::new(&chords))
     }
+
+    pub fn parse(string: &str) -> Result<Self> {
```

Comment:
Per above comment, I liked the idea of exposing a corresponding `parse()` method on these structs with `parser()` methods. I wanted to picture it as a trait that implements both `parser()` + `parse()` (with a default implementation of `parse()` that invokes `parser()`) but I ran into the fact that I guess a trait method can't return an eg `impl Parser`, so I wasn't sure what to do about that

Reply:
We could change those `impl Parser`s into `Box<Parser<ReturnType>`, I think? That seems like a reasonable trait to have.

Reply:
Alternatively, you could code generate it with a macro 😈

Reply:
Ok opened #12